libshaderc_spvc/src/spvcir_pass.cc

```diff
@@ -23,25 +23,46 @@ SpvcIrPass::SpvcIrPass(spirv_cross::ParsedIR *ir) {
   ir_ = ir;
   current_function_ = nullptr;
   current_block_ = nullptr;
+  status_ = Status::SuccessWithoutChange;
 
   auto s = ir_->spirv.data();
-  assert(ir_->spirv.size() > 3 && "spirv data is too small");
+  CheckAndSetErrorMessage(ir_->spirv.size() > 3,
+                          "SpvcIrPass: spirv data is too small");
   uint32_t bound = s[3];
   ir_->set_id_bounds(bound);
 }
 
+void SpvcIrPass::CheckAndSetErrorMessage(bool condition, std::string message) {
+  if (!condition) {
+    status_ = Status::Failure;
+    if (consumer()) {
+      consumer()(SPV_MSG_ERROR, 0, {0, 0, 0}, message.c_str());
+    } else {
+      assert(condition && message.c_str());
+    }
+  }
+}
+
 Pass::Status SpvcIrPass::Process() {
```

Comment:
If you don't change the constructor, you need to check that status_ isn't Failure at the top of this function.